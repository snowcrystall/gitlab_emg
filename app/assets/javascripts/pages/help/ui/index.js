import initUIKit from '~/ui_development_kit';

initUIKit();
