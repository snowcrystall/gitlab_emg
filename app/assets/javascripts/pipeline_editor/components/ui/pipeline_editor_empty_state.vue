<script>
import { GlButton, GlSprintf } from '@gitlab/ui';
import { __ } from '~/locale';
import PipelineEditorFileNav from '~/pipeline_editor/components/file_nav/pipeline_editor_file_nav.vue';
import glFeatureFlagsMixin from '~/vue_shared/mixins/gl_feature_flags_mixin';

export default {
  components: {
    GlButton,
    GlSprintf,
    PipelineEditorFileNav,
  },
  i18n: {
    title: __('Optimize your workflow with CI/CD Pipelines'),
    body: __(
      'Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
    btnText: __('Create new CI/CD pipeline'),
  },
  mixins: [glFeatureFlagsMixin()],
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
  },
  computed: {
    showFileNav() {
      return this.glFeatures.pipelineEditorBranchSwitcher;
    },
    showCTAButton() {
      return this.glFeatures.pipelineEditorEmptyStateAction;
    },
  },
  methods: {
    createEmptyConfigFile() {
      this.$emit('createEmptyConfigFile');
    },
  },
};
</script>
<template>
  <div>
    <pipeline-editor-file-nav v-if="showFileNav" v-on="$listeners" />
    <div class="gl-display-flex gl-flex-direction-column gl-align-items-center gl-mt-11">
      <img :src="emptyStateIllustrationPath" />
      <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
      <p class="gl-mt-3">
        <gl-sprintf :message="$options.i18n.body">
          <template #code="{ content }">
            <code>{{ content }}</code>
          </template>
        </gl-sprintf>
      </p>
      <gl-button
        v-if="showCTAButton"
        variant="confirm"
        class="gl-mt-3"
        @click="createEmptyConfigFile"
      >
        {{ $options.i18n.btnText }}
      </gl-button>
    </div>
  </div>
</template>
