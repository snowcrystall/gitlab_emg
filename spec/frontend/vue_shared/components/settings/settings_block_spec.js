import { GlButton } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import component from '~/vue_shared/components/settings/settings_block.vue';

describe('Settings Block', () => {
  let wrapper;

  const mountComponent = (propsData) => {
    wrapper = shallowMount(component, {
      propsData,
      slots: {
        title: '<div data-testid="title-slot"></div>',
        description: '<div data-testid="description-slot"></div>',
        default: '<div data-testid="default-slot"></div>',
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  const findDefaultSlot = () => wrapper.find('[data-testid="default-slot"]');
  const findTitleSlot = () => wrapper.find('[data-testid="title-slot"]');
  const findDescriptionSlot = () => wrapper.find('[data-testid="description-slot"]');
  const findExpandButton = () => wrapper.find(GlButton);

  it('renders the correct markup', () => {
    mountComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('has a default slot', () => {
    mountComponent();

    expect(findDefaultSlot().exists()).toBe(true);
  });

  it('has a title slot', () => {
    mountComponent();

    expect(findTitleSlot().exists()).toBe(true);
  });

  it('has a description slot', () => {
    mountComponent();

    expect(findDescriptionSlot().exists()).toBe(true);
  });

  describe('slide animation behaviour', () => {
    it('is animated by default', () => {
      mountComponent();

      expect(wrapper.classes('no-animate')).toBe(false);
    });

    it.each`
      slideAnimated | noAnimatedClass
      ${true}       | ${false}
      ${false}      | ${true}
    `(
      'sets the correct state when slideAnimated is $slideAnimated',
      ({ slideAnimated, noAnimatedClass }) => {
        mountComponent({ slideAnimated });

        expect(wrapper.classes('no-animate')).toBe(noAnimatedClass);
      },
    );
  });

  describe('expanded behaviour', () => {
    it('is collapsed by default', () => {
      mountComponent();

      expect(wrapper.classes('expanded')).toBe(false);
    });

    it('adds expanded class when the expand button is clicked', async () => {
      mountComponent();

      expect(wrapper.classes('expanded')).toBe(false);
      expect(findExpandButton().text()).toBe('Expand');

      await findExpandButton().vm.$emit('click');

      expect(wrapper.classes('expanded')).toBe(true);
      expect(findExpandButton().text()).toBe('Collapse');
    });

    it('is expanded when `defaultExpanded` is true no matter what', async () => {
      mountComponent({ defaultExpanded: true });

      expect(wrapper.classes('expanded')).toBe(true);

      await findExpandButton().vm.$emit('click');

      expect(wrapper.classes('expanded')).toBe(true);

      await findExpandButton().vm.$emit('click');

      expect(wrapper.classes('expanded')).toBe(true);
    });
  });
});
