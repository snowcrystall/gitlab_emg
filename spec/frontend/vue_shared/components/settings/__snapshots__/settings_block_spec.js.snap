// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Block renders the correct markup 1`] = `
<section
  class="settings"
>
  <div
    class="settings-header"
  >
    <h4>
      <div
        data-testid="title-slot"
      />
    </h4>
     
    <gl-button-stub
      buttontextclasses=""
      category="primary"
      icon=""
      size="medium"
      variant="default"
    >
      
      Expand
    
    </gl-button-stub>
     
    <p>
      <div
        data-testid="description-slot"
      />
    </p>
  </div>
   
  <div
    class="settings-content"
  >
    <div
      data-testid="default-slot"
    />
  </div>
</section>
`;
